app/services/rummager_saver.rb

```diff
@@ -39,16 +42,22 @@ def destroy(action: :delete)
 private
 
   def update_elasticsearch(query, action)
-    if action != :delete && query.bets.any?
-      es_doc = ElasticSearchBet.new(query)
-      Services.rummager.add_document(es_doc.id, es_doc.body, 'metasearch')
-    elsif action != :create # so hitting save with no best bets will delete the query
-      es_doc_id = ElasticSearchBetIDGenerator.generate(query.query, query.match_type)
-      Services.rummager.delete_document(URI.escape(es_doc_id), 'metasearch') # rubocop:disable Lint/UriEscapeUnescape
+    if %i[update create].include?(action) && query.bets.any? # prevents queries with no bets being indexed
+      add_to_elasticsearch
+    elsif action == :delete || action == :update && query.bets.none? # removes queries with no bets from the index
+      remove_from_elasticsearch
+    elsif action == :update_bets # removing the final bet will de-index the query, removing any others will re-index
+      query.bets.any? ? add_to_elasticsearch : remove_from_elasticsearch
```

Comment:
I wonder if it's worth adding an `else` here which raises an exception if you pass in an `action` that isn't one we expect.

Reply:
good idea, I've added error handling. It makes the overall if/else slightly more complicated - but there lots of different situations being managed here.